src/golang/lib/airflow/utils.go

```diff
@@ -5,11 +5,35 @@ import (
 )
 
 // generateDagId generates an Airflow DAG ID for a workflow.
-func generateDagId(workflowName string) string {
-	return strings.ReplaceAll(workflowName, " ", "_")
+func generateDagId(workflowName string) (string, error) {
+	return prepareId(workflowName)
 }
 
 // generateTaskId generates an Airflow task ID for an operator.
-func generateTaskId(operatorName string) string {
-	return strings.ReplaceAll(operatorName, " ", "_")
+func generateTaskId(operatorName string) (string, error) {
+	return prepareId(operatorName)
+}
+
+// prepareId replaces all non-alphanumeric characters in `s` with
```

Comment:
This must have been a pain to do. Are there other examples of projects converting operators into airflow tasks that also do this?

Reply:
Hmm I actually haven't come across other projects generating Airflow scripts